node/impl/full/eth_events.go

```diff
@@ -118,35 +118,43 @@ func ethFilterResultFromEvents(ctx context.Context, evs []*filter.CollectedEvent
 			return nil, xerrors.Errorf("emitter to addr: %w", err)
 		}
 
-		actor, err := sa.StateGetActor(ctx, emitterAddr, ev.TipSetKey)
+		ts, err := sa.Chain.GetTipSetFromKey(ctx, ev.TipSetKey)
+		if err != nil {
+			return nil, xerrors.Errorf("loading tipset: %w", err)
+		}
+		stateRoot, _, err := sa.StateManager.TipSetState(ctx, ts)
+		if err != nil {
+			return nil, xerrors.Errorf("loading tipset state: %w", err)
+		}
+		actor, err := sa.StateManager.LoadActorRaw(ctx, emitterAddr, stateRoot)
```

Comment:
this is the relevant section that I'm testing out

The problem with `sa.StateGetActor(tsk)` is that it it loads `ParentState` but the event may arise from a tipset where the actor id was first introduced, so we fail with an "unknown actor" error. But if we fast-forward by one then it's there.